routes/fire_weather.py

```diff
@@ -0,0 +1,753 @@
+import asyncio
+import logging
+import xarray as xr
+import io
+import numpy as np
+from flask import Blueprint, render_template, request
+import datetime
+import time
+
+# local imports
+from generate_urls import generate_wcs_query_url
+from generate_requests import generate_wcs_getcov_str, generate_netcdf_wcs_getcov_str
+from fetch_data import (
+    fetch_data,
+    describe_via_wcps,
+    ymd_to_cftime_value,
+    cftime_value_to_ymd,
+    get_encoding_from_axis_attributes,
+    get_attributes_from_time_axis,
+    get_poly,
+    fetch_bbox_netcdf,
+)
+from validate_request import (
+    latlon_is_numeric_and_in_geodetic_range,
+    check_geotiffs,
+    check_poly_in_geotiffs,
+    validate_year,
+    validate_var_id,
+)
+from zonal_stats import (
+    get_scale_factor,
+    rasterize_polygon,
+    interpolate,
+    calculate_zonal_means_vectorized,
+)
+from csv_functions import create_csv
+from luts import summer_fire_danger_ratings_dict
+
+from . import routes
+
+logger = logging.getLogger(__name__)
+
+fire_weather_api = Blueprint("fire_weather_api", __name__)
+
+
+#### SETUP AND METADATA ####
+
+# we have one geotiff to validate all fire weather variables, so its not the same as coverage names
+fire_weather_geotiff = "cmip6_all_fire_weather_variables"
+
+fire_weather_coverage_ids = [
+    "cmip6_bui",
+    "cmip6_dc",
+    "cmip6_dmc",
+    "cmip6_ffmc",
+    "cmip6_fwi",
+    "cmip6_isi",
+]
+
+
+# get the basic metadata for all fire weather coverages
+async def get_coverage_metadata(coverage_id):
+    """Get the coverage metadata."""
+    metadata = await describe_via_wcps(coverage_id)
+    return metadata
+
+
+var_coverage_metadata = {}
+for coverage_id in fire_weather_coverage_ids:
+    coverage_metadata = asyncio.run(get_coverage_metadata(coverage_id))
+    base_date, time_min, time_max = get_attributes_from_time_axis(coverage_metadata)
+    # below assumes only one variable per coverage
+    var_coverage_metadata[list(coverage_metadata["metadata"]["bands"].keys())[0]] = {
+        "coverage_id": coverage_id,
+        "model_encoding": get_encoding_from_axis_attributes("model", coverage_metadata),
+        "start_cf_time": time_min,  # integer days since base date
+        "end_cf_time": time_max,  # integer days since base date
+        "base_date": base_date,  # datetime.datetime object
+        "start_date": cftime_value_to_ymd(
+            time_min, base_date
+        ),  # (year, month, day) tuple
+        "end_date": cftime_value_to_ymd(
+            time_max, base_date
+        ),  # (year, month, day) tuple
+    }
+
+
+#### VALIDATION FUNCTIONS ####
+
+
+def validate_latlon(lat, lon):
```

Comment:
I like this wrapper, but there is a function with this exact name in `validate_request.py` so this has some potential to be a trap. I'd suggest renaming this function and/or pushing this to the validate_request module where other endpoints could use this wrapper.